pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/invertedindex/RealtimeLuceneTextIndex.java

```diff
@@ -122,12 +126,21 @@ public MutableRoaringBitmap getDocIds(String searchQuery) {
     Callable<MutableRoaringBitmap> searchCallable = () -> {
       IndexSearcher indexSearcher = null;
       try {
-        QueryParser parser = new QueryParser(_column, _analyzer);
+        // Lucene Query Parser is JavaCC based. It is stateful and should
+        // be instantiated per query. Analyzer on the other hand is stateless
+        // and can be created upfront.
+        QueryParserBase parser = _queryParserClassConstructor.newInstance(_column, _analyzer);
```

Comment:
Why do we need this change? The comment above seems not explaining this line.

Reply:
Should we also do this on the non realtime lucene index too?

Reply:
I believe the "non realtime lucene index" code path is in `LuceneTextIndexReader.java` which has the same code added